client/me/purchases/index.js

```diff
@@ -14,87 +14,96 @@ import meController from 'me/controller';
 import { siteSelection } from 'my-sites/controller';
 import controller from './controller';
 import * as paths from './paths';
-import { makeLayout, render as clientRender } from 'controller';
+import { makeLayout, redirectLoggedOut, render as clientRender } from 'controller';
 
-export default function() {
+export default function( router ) {
 	if ( config.isEnabled( 'manage/payment-methods' ) ) {
-		page(
+		router(
 			paths.addCreditCard,
+			redirectLoggedOut,
 			meController.sidebar,
 			controller.addCreditCard,
 			makeLayout,
 			clientRender
 		);
 
 		// redirect legacy urls
-		page( '/payment-methods/add-credit-card', () => page.redirect( paths.addCreditCard ) );
+		router( '/payment-methods/add-credit-card', () => page.redirect( paths.addCreditCard ) );
 	}
 
-	page(
+	router(
 		paths.billingHistory,
+		redirectLoggedOut,
 		meController.sidebar,
 		billingController.billingHistory,
 		makeLayout,
 		clientRender
 	);
 
-	page(
+	router(
 		paths.billingHistoryReceipt(),
+		redirectLoggedOut,
 		meController.sidebar,
 		billingController.transaction,
 		makeLayout,
 		clientRender
 	);
 
-	page( paths.purchasesRoot, meController.sidebar, controller.list, makeLayout, clientRender );
+	router( paths.purchasesRoot, meController.sidebar, controller.list, makeLayout, clientRender );
```

Comment:
I think you need to add the `redirectLoggedOut` here too.  Otherwise it doesn't work on the main `me/purchases` root URL itself.

Reply:
Good point. Thanks!